compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -2067,7 +2067,14 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
 
                 // Don't print the tuple of capture types
                 if !is_upvar_tys_infer_tuple {
-                    err.note(&format!("required because it appears within the type `{}`", ty));
+                    let msg = format!("required because it appears within the type `{}`", ty);
+                    match ty.kind() {
+                        ty::Adt(def, _) => match self.tcx.item_name_from_hir(def.did) {
```

Comment:
Why use `item_name_from_hir` instead of `opt_item_name`? Do you expect `def` to always be local?

Reply:
Using `item_name_from_hir` isn't going to cause ICEs with non-local definitions, we'll only be missing the span in those cases. I assume the idea is that this span is only helpful and is actionable for locally defined types?

Reply:
It feels to me that seeing where the requirement is included in another crate isn't as useful as when the requirement comes from the current crate, because people can't change them, but it might useful for extra context? 🤷‍♀️

I'll make the change and see what it actually looks like in the test suite.

Reply:
Just included this change. No effects in the tests (but might increase verbosity in the wild). Regardless, I can change it after real world usage if people complain.